compose/queue/api_data/taskqueue_v1beta2.py

```diff
@@ -0,0 +1,626 @@
+        taskqueue: A string of the affected taskqueue.
+
+    POST:
+        Acquires a lease on the topmost N unowned tasks in the
+        specified queue.
+
+        Query:
+            leaseSecs: An integer of the number of seconds to lease
+                the tasks for.
+            numTasks: An integer of the number of tasks to lease.
+            groupByTag: (Optional) true or false, determining whether
+                to get tasks by tag.
+            tag: (Optional) A string specifing which tag leased tasks
+                must have.
+        Returns:
+            A json object containing a kind and items attribute.
+            Where items is a list of tasks.
+    """
+    if project is None:
+        return "You must specify a project.", 400
+    elif taskqueue is None:
+        return "You must specify a taskqueue.", 400
+
+    project = project.replace("b~", "")
+    queue_key = ".".join([project, taskqueue])
+
+    # Acquires a lease on the topmost N unowned tasks in the specified
+    # queue.
+    #     Required query parameters: leaseSecs, numTasks
+    if request.method == "POST":
+        leaseSecs = request.args.get("leaseSecs")
+        numTasks = request.args.get("numTasks")
+        if leaseSecs is None or numTasks is None:
+            return "Missing leaseSecs and numTasks in query.", 400
+
+        groupByTag = request.args.get("groupByTag", False)
+        tag = request.args.get("tag", "")
+        if not groupByTag and tag != "":
+            return "In query tag specified without groupByTag.", 400
+
+        start = 0
+        tasks = []
+        now_milliseconds = now()
+        numTasks = min(int(numTasks), 1000)
+        while len(tasks) < numTasks:
+            tasks_needed = numTasks - len(tasks)
+            end = start + tasks_needed
+            keys = r.zrange(name=queue_key, start=start, end=end)
+            if len(keys) == 0:
+                break
+
+            for key in keys:  # Could become parallelizable
+                task_string = r.get(name=key)
+                if task_string is None:
+                    continue
+
+                task_string = task_string.decode()
+                task_json = json.loads(task_string)
+                task = Task.from_json(task_json)
+
+                if groupByTag and task.tag != tag:
+                    continue
+
+                if task.leaseTimestamp > now_milliseconds:
+                    continue
+
+                task.leaseTimestamp = now_milliseconds + int(leaseSecs) * 10**6
+                task_json = task.to_json()
+                tasks.append(task_json)
+
+                task.retry_count += 1
+                task_string = json.dumps(task_json)
+                r.set(name=key, value=task_string)
+
+            start = end + 1
+
+        return json.dumps({
+            "kind": "taskqueues#tasks",
+            "items": tasks
+        })
+
+
+@taskqueue_v1beta2_api.route(
+    "/<project>/taskqueues/<taskqueue>/tasks/<task_id>",
```

Comment:
Its a REST API url, this doesn't get deployed and is documented by Google there is no hiding it.